### PR TITLE
A `scheduler` replica

### LINKED ISSUE #44: Make a replica of a `Scheduler` class that delivers all function call request to the artiq server
### Feature you want to implement
As mentioned in [Issue#96](https://github.com/snu-quiqcl/iquip/issues/96) of `iquip`, I need a port to send the modify request of the queued experiment to the _real_ scheduler in the server.

### How the feature is implemented
In `main.py`, I will make a POST (maybe GET if it needs return value) request path that receives a function call request of functions in [`artiq/artiq/master/scheduler.py`](https://github.com/m-labs/artiq/blob/master/artiq/master/scheduler.py#L400) and calls the functions in scheduler that is connected by `remote = get_client("master_schedule")`.

### PR DESCRIPTION
**Pull request by giwon2004**
It is a simple replica of a `scheduler` in [`artiq/master/scheduler.py`](https://github.com/m-labs/artiq/blob/master/artiq/master/scheduler.py#L400) to receive signals to delete experiments and request termination of experiments from our GUI.

There are two functions implemented:
`delete_experiment(rid)`: Kills the run with the specified RID.
`request_termination_of_experiment(rid)`: Requests graceful termination of the run with the specified RID.

Both are already defined as a method of `Scheduler` instance in [`artiq/master/scheduler.py`](https://github.com/m-labs/artiq/blob/master/artiq/master/scheduler.py#L400) and above two calls each one.

This closes #44.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -108,6 +108,28 @@ async def get_experiment_code(file: str) -> str:
     return code
 
 
+@app.post("/experiment/delete/")
+async def delete_experiment(rid):
```
Comment (BECATRUE) [inline main.py]: Please note the type hint.

File: main.py [hunk 2]
```diff
@@ -108,6 +108,28 @@ async def get_experiment_code(file: str) -> str:
...
+    """Kills the run with the specified RID.
+
+    Args:
+        rid: rid value of the target experiment.
```
Comment (BECATRUE) [inline main.py]: How about "The run identifier of the ..."?

File: main.py [hunk 3]
```diff
@@ -108,6 +108,28 @@ async def get_experiment_code(file: str) -> str:
...
+    """
+    remote = get_client("master_schedule")
+    remote.delete(rid)
+
+
+@app.post("/experiment/request_termination/")
```
Comment (BECATRUE) [inline main.py]: How about "/experiment/terminate/"?
It looks too long.

Comment (BECATRUE) [reply main.py:122]: Umm.. Why do you need this? I think `remote.delete(rid)` is enough..

Is there any reason?

**Review by BECATRUE — CHANGES_REQUESTED** [review]

File: main.py (hunk 3 above)
Comment (giwon2004) [reply main.py]: The original `scheduler` supports these two different functions, and I guess there is a difference like one between Alt+F4 and quitting by clicking the X button.

Comment (BECATRUE) [reply main.py]: What is difference betwwen `delete()` and `request_termination()`?

Comment (giwon2004) [reply main.py]: I do not know in detail, but I made a _replica_ so I thought I need to make them for future use.

Comment (BECATRUE) [reply main.py]: @kangz12345 How about you?

**Comment by BECATRUE** [comment]
Also, I think `get` is more appropriate even if it doesn't need to return something, because it doesn't create or change the server.

File: main.py (hunk 3 above)
Comment (kangz12345) [reply main.py]: See also: [source code](https://m-labs.hk/artiq/manual/_modules/artiq/master/scheduler.html#Scheduler.delete)

`request_termination()` will do some proper works before deleting an experiment.
I think both are required (`delete()` could be like force-quit).

**Comment by kangz12345** [comment]
[RFC9110](https://www.rfc-editor.org/rfc/rfc9110#name-method-definitions) would be helpful to decide which method to use. Moreover, it would be a good practice for us to read the whole document.

**Comment by giwon2004** [comment]
I guess `post` is enough. It is better for security too.

**Review by BECATRUE — DISMISSED** [review]
LGTM!

Did you check if both requests work well?

**Review by kangz12345 — DISMISSED** [review]

**Comment by BECATRUE** [comment]
@giwon2004 Oh, because of merging #36 to `develop` branch, the conflicts occurs.
Please fix this.

**Comment by kangz12345** [comment]
You can just click the "Resolve conflicts" button and remove "====" or "<<<<" things to make the final version.

**Comment by giwon2004** [comment]
> You can just click the "Resolve conflicts" button and remove "====" or "<<<<" things to make the final version.

I clicked **Mark as resolved**, but the conflict did not resolve.
So I clicked **Merge commit**. Is this what you meant to do?

**Comment by BECATRUE** [comment]
> > You can just click the "Resolve conflicts" button and remove "====" or "<<<<" things to make the final version.
> 
> I clicked **Mark as resolved**, but the conflict did not resolve. So I clicked **Merge commit**. Is this what you meant to do?

Probably you are right! Then, I will leave reviews again!

File: main.py [hunk 4]
```diff
@@ -97,6 +97,41 @@ async def get_experiment_info(file: str) -> Any:
     return remote.examine(file)
 
 
+@app.get("/experiment/code/")
+async def get_experiment_code(file: str) -> str:
```
Comment (BECATRUE) [inline main.py]: Umm.. Why was this code created?

If you handled this during resolving the conflicts, then please remove this function!

**Review by BECATRUE — CHANGES_REQUESTED** [review]

**Review by BECATRUE — APPROVED** [review]
LGTM!

**Review by kangz12345 — APPROVED** [review]

**Pull request merged**